src/models/multi_modal_vision_model.cpp

```diff
@@ -104,41 +30,19 @@ int64_t GetNumImageTokens(const std::vector<GeneratorParams::Input>& extra_input
     return 0;
   }
 
-  if (image_sizes->ort_tensor_->GetTensorTypeAndShapeInfo()->GetShape() != std::vector<int64_t>{1, 2}) {
-    throw std::runtime_error("image_sizes tensor must have 2 elements");
+  auto num_images = pixel_values->ort_tensor_->GetTensorTypeAndShapeInfo()->GetShape()[0];
+  if (image_sizes->ort_tensor_->GetTensorTypeAndShapeInfo()->GetShape() != std::vector<int64_t>{num_images, 2}) {
+    throw std::runtime_error("image_sizes tensor must be of shape (num_images, 2)");
   }
 
   auto image_sizes_data = image_sizes->ort_tensor_->GetTensorMutableData<int64_t>();
-  const int64_t h = image_sizes_data[0] / 336;
-  const int64_t w = image_sizes_data[1] / 336;
-  return ((h * w + 1) * 144) + 1 + ((h + 1) * 12);
-}
-
-std::unique_ptr<OrtValue> GetVisualFeatures(OrtAllocator& device_allocator, const SessionInfo& session_info,
-                                            const std::string& visual_features_name, int32_t hidden_size,
-                                            int64_t num_image_tokens) {
-  constexpr int32_t batch_size = 1;
-  if (!session_info.HasOutput(visual_features_name)) {
-    throw std::runtime_error("Visual features output not found in the model");
+  int64_t num_image_tokens = 0;
+  for (int i = 0; i < num_images; i++) {
+    int64_t h = image_sizes_data[i * num_images] / 336;
+    int64_t w = image_sizes_data[i * num_images + 1] / 336;
+    num_image_tokens += static_cast<int64_t>((h * w + 1) * 144) + 1 + ((h + 1) * 12);
```

Comment:
I don't think we need the static_cast here since h and w and num_image_tokens are all int64_t.

Reply:
Removed